### DISCUSSION TITLE
A question about setting the integration step

### DISCUSSION THREAD

**Discussion by ujust4fun** [JSBSim-Team/jsbsim#851] [Q&A]
I met a problem with setting the integration step. 

Expected your help!

**Comment by ujust4fun** [comment]
I met a question when I set the different time-step. 
```python
import jsbsim
import matplotlib.pyplot as plt
import print

fdm = jsbsim.FGFDMExec(None)  # Use JSBSim default aircraft data.
fdm.load_model('737')
fdm.run_ic()
fdm['propulsion/engine[0]/set-running']=1
fdm['propulsion/engine[1]/set-running']=1
fdm['fcs/throttle-cmd-norm']=1

fdm["fcs/left-brake-cmd-norm"] = 0
fdm["fcs/right-brake-cmd-norm"] = 0
fdm["fcs/brake-parking-cmd-norm"] = 0
fdm.set_dt(0.04)
mach=[]
time=[]
while fdm.run() and fdm.get_sim_time()<50:
    time.append(fdm.get_sim_time())
    print('Sim time:',fdm.get_sim_time())
    print(fdm["position/h-sl-meters"])
    mach.append(fdm['velocities/mach'])
plt.plot(time,mach,label='dt=0.04')
fdm.reset_to_initial_conditions(1)
fdm.run_ic()
fdm['propulsion/engine[0]/set-running']=1
fdm['propulsion/engine[1]/set-running']=1
fdm['fcs/throttle-cmd-norm']=1

fdm["fcs/left-brake-cmd-norm"] = 0
fdm["fcs/right-brake-cmd-norm"] = 0
fdm["fcs/brake-parking-cmd-norm"] = 0
fdm.set_dt(0.08)
mach2=[]
time2=[]
while fdm.run() and fdm.get_sim_time()<50:
    time2.append(fdm.get_sim_time())
    print('Sim time:',fdm.get_sim_time())
    print(fdm["position/h-sl-meters"])
    mach2.append(fdm['velocities/mach'])
plt.plot(time2,mach2,label='dt=0.08')
plt.legend()
plt.show()
```

the result is different caused by the different **set_time(0.04) set_time(0.08)**.  I am confused by this. Could you explain it to me？ thx
![Figure_1](https://user-images.githubusercontent.com/25298124/222966595-142622b9-6ee9-4bfe-b374-0f3dd6fe318f.png)

**Reply by ujust4fun** [reply]
By the way, `fdm.reset_to_initial_conditions(1)`  is a reset function, but what the parameter means. I am confused by its value explanation

**Reply by bcoconni** [reply]
You are calling `fdm.run_ic()` without having properly set the initial conditions first.
If I had `fdm['ic/h-sl-ft'] = 4.0` to your script:
```python
fdm = jsbsim.FGFDMExec(None)
fdm.load_model('737')
fdm['ic/h-sl-ft'] = 4.0  # Spawn the aircraft 4ft above the ground.
fdm.run_ic()
```
By default, JSBSim uses an initial height of 0ft and that results in the landing gears being compressed by almost 4ft. When the simulation starts, the compression energy in the landing gears is released and the aircraft is unrealistically propelled to the outer space (more details are given in the discussion #819 albeit your case is a variation as you don't hit the ground but initiates the simulation as if the aircraft had just violently hit the ground).

Once I fixed the initial height parameter, I'm getting something that is more plausible

![image](https://user-images.githubusercontent.com/7502904/227497043-51fa0bb5-1586-4d9d-b6ed-fee212e10b09.png)

Of course, relaxing the time step results in a lower fidelity simulation.

**Reply by bcoconni** [reply]
> By the way, `fdm.reset_to_initial_conditions(1)` is a reset function, but what the parameter means. I am confused by its value explanation

The value is meant to open new output files upon reset or continue logging data in the same output files: 
* Setting the parameter to `1` means that the current output files will be closed and new output files (with their name appended with a counter `_1`, `_2`, etc.) will be used.
* Setting the parameter to `0` means that the simulation will continue logging its output in the same output files than before calling `fdm.reset_to_initial_conditions`.

**Comment by ujust4fun** [comment]
Still have a question when i want to output the data from the 737.xml.  
I have set the output code like follows,
```python
fdm.set_output_filename(0, 'this_one.csv')
fdm.enable_output()
while fdm.run() and fdm.get_sim_time()<50:
    time2.append(fdm.get_sim_time())
    h2.append(fdm["position/h-sl-meters"])
    mach2.append(fdm['velocities/mach'])
plt.plot(time2,h2,label='dt=0.08')
plt.legend()
plt.show()
```
and also ,  I copied the output property from the c172x.xml to  737.xml  and commented the other output method in latter. 
```xml
    <output name="JSBout777.csv" type="CSV" rate="10">
        <rates> ON             </rates>
        <velocities> ON        </velocities>
        <position> ON          </position>
        <atmosphere> OFF       </atmosphere>
        <fcs> ON               </fcs>
        <ground_reactions> OFF </ground_reactions>
        <propulsion> ON       </propulsion>
        <simulation> ON        </simulation>
        <massprops> ON        </massprops>
        <forces> OFF           </forces>
        <moments> OFF          </moments>
        <aerosurfaces> OFF      </aerosurfaces>
        <property> position/vrp-gc-latitude_deg </property>
        <property> position/vrp-longitude_deg   </property>
        <property> position/vrp-radius-ft       </property>
        <function name="velocities/pi-deg_sec">
          <todegrees> <p> velocities/pi-rad_sec </p> </todegrees>
        </function>
    </output>
```
However, there is none output . Where did i wrong? hope you can help me.

**Reply by bcoconni** [reply]
Output files are written to the output path that can be accessed using the following command:
```python
fdm.get_output_path()
```
So your file should be located in the folder returned by this command.

If you need output files to be written elsewhere, use `fdm.set_output_path()`

Also to rename the output file from `JSBout777.csv` to `this_one.csv`, the command `fdm.set_output_filename()` should be called before `fdm.run_ic()` or it will be ignored. And you don't need to call `fdm.enable_output()` as it is enabled by default.

The first lines of your code should look something like:
```python
fdm = jsbsim.FGFDMExec(None)
fdm.load_model('737')
fdm['ic/h-sl-ft'] = 4.0  # Spawn the aircraft 4ft above the ground.
fdm_set_output_path('folder/for/output/files')
fdm.set_output_filename(0, 'this_one.csv')
fdm.run_ic()
```
